src/adaptors/tcp_lite/tcp_lite.c

```diff
@@ -1410,10 +1418,12 @@ static void CORE_activate(void *context, qdr_connection_t *core_conn)
 
     case TL_CONNECTION:
         conn = (tcplite_connection_t*) common;
+        sys_mutex_lock(&conn->activation_lock);
         if (IS_ATOMIC_FLAG_SET(&conn->raw_opened)) {
             SET_ATOMIC_FLAG(&conn->core_activation);
```

Comment:
With this lock in place, are there any atomic flags we can convert to bools?

Reply:
It looks like the `conn->raw_opened` flag can be knocked down to boolean. The `conn->core_activation` flag needs to remain atomic.

Reply:
I have added code to this PR that set the `conn->raw_opened` to boolean. Please take a look.

Reply:
After further analysis, the atomic flags that are used inside the new activation lock are also sometimes used outside the activation lock, so my conclusion is that those flags cannot be converted from atomic to boolean